src/controllers/forumController.js

```diff
@@ -0,0 +1,68 @@
+const Forum = require('../models/Forum');
+const { StatusCodes } = require('http-status-codes');
+const CustomError = require('../errors');
+
+const createRoom = async (req, res) => {
+  try {
+    const { name } = req.body;
+
+    if (!name) {
+      throw new CustomError.BadRequestError('Room name is required');
+    }
+
+    const existingRoom = await Forum.findOne({ name });
+
+    if (existingRoom) {
+      throw new CustomError.ConflictError('Room with this name already exists');
+    }
+
+    const newRoom = await Forum.create({ name });
+    res.status(StatusCodes.CREATED).json(newRoom);
+  } catch (error) {
+    console.error(error);
+    res
+      .status(error.statusCode || StatusCodes.INTERNAL_SERVER_ERROR)
+      .json({ error: error.message });
+  }
```

Comment:
try/catch shouldn't be needed here. Any errors raised in the express controllers will be handled by the `error-handler` middleware thanks to our use of the `express-async-errors` library; and that middleware already does the `console.error` and `res.status` bits that you've duplicated here.